Should add SCRAM client to vertx-sql-client dependency list

# pg_hba.conf

```
host    all             all             192.168.20.72/32        scram-sha-256
```

# Error Log

```
java.lang.NoClassDefFoundError: com/ongres/scram/common/stringprep/StringPreparation
        at io.vertx.pgclient.impl.codec.InitCommandCodec.handleAuthenticationSasl(InitCommandCodec.java:60)
        at io.vertx.pgclient.impl.codec.PgDecoder.decodeAuthentication(PgDecoder.java:357)
        at io.vertx.pgclient.impl.codec.PgDecoder.decodeMessage(PgDecoder.java:139)
        at io.vertx.pgclient.impl.codec.PgDecoder.channelRead(PgDecoder.java:111)
        at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:251)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
        at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:719)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:655)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:581)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.ClassNotFoundException: com.ongres.scram.common.stringprep.StringPreparation
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
        ... 21 more
```

# Solution: pom.xml

```
    <dependency>
      <groupId>com.ongres.scram</groupId>
      <artifactId>client</artifactId>
      <version>2.1</version>
    </dependency>
    <dependency>
      <groupId>io.vertx</groupId>
      <artifactId>vertx-pg-client</artifactId>
      <version>4.0.3</version>
    </dependency>
```

--- a/vertx-pg-client/pom.xml
+++ b/vertx-pg-client/pom.xml
@@ -54,7 +54,6 @@
       <groupId>com.ongres.scram</groupId>
       <artifactId>scram-client</artifactId>
       <version>3.1</version>
-      <optional>true</optional>
     </dependency>
 
     <!-- Testing purposes -->

--- a/vertx-pg-client/src/main/asciidoc/index.adoc
+++ b/vertx-pg-client/src/main/asciidoc/index.adoc
@@ -233,28 +233,9 @@ $ PGUSER=user \
 
 === SASL SCRAM-SHA-256 authentication mechanism.
 
-To use the SASL `SCRAM-SHA-256` authentication add the following dependency to the _dependencies_ section of your build descriptor:
+The `vertx-pg-client` supports the `SCRAM-SHA-256` and `SCRAM-SHA-256-PLUS` SASL authentication. From version v5.0.0 on `vertx-pg-client` ships with the `com.ongres.scram` library dependency, when migrating from an older version `vertx-pg-client` users can remove the `com.ongres.scram` library from their `pom.xml` and `build.gradle` files.
 
-* Maven (in your `pom.xml`):
-
-[source,xml]
-----
-<dependency>
-  <groupId>com.ongres.scram</groupId>
-  <artifactId>scram-client</artifactId>
-  <version>3.1</version>
-</dependency>
-----
-* Gradle (in your `build.gradle` file):
-
-[source,groovy]
-----
-dependencies {
-  compile 'com.ongres.scram:scram-client:3.1'
-}
-----
-
-When the database requires a scram authentication and the scram client jar is not on the class/module path, the connection will be closed by the client.
+If a client uses other auth methods only then the transitive `com.ongres.scram` dependency can be excluded using maven `<exclusions>` or gradle `exclude` to reduce the software size.
 
 include::queries.adoc[leveloffset=1]
 

